atintegrators/gwigR.c

```diff
@@ -0,0 +1,477 @@
+    if ( abs(kx/kw) > GWIG_EPS ) {
+      sxkx = sin(kx * x)/kx;	
+    } else {
+      sxkx = x*sinc(kx*x);
+    }
+
+    axpy = axpy + pWig->HCw[i]*(kw/kz)*ky*sxkx*shy*sz;
+  }
+
+
+  /* Vertical Wiggler: note that one potentially could have: ky=0 */
+  for (i = 0; i < pWig->NVharm; i++ ) {
+    pWig->VCw[i] = pWig->VCw_raw[i]*(pWig->Aw)/(gamma0*beta0);
+    kx = pWig->Vkx[i];
+    ky = pWig->Vky[i];
+    kz = pWig->Vkz[i];
+    tz = pWig->Vtz[i];
+
+    shx = sinh(kx * x);
+    sy  = sin(ky * y);
+    sz  = sin(kz * z + tz);
+    ax  = ax + pWig->VCw[i]*(kw/kz)*(ky/kx)*shx*sy*sz;
+
+    chx = cosh(kx * x);
+    cy  = cos(ky * y);
+    axpy = axpy + pWig->VCw[i]*(kw/kz)* pow(ky/kx,2) *chx*cy*sz;      
+  }
+
+  *pax   = ax;
+  *paxpy = axpy;
+}
+
+
+void GWigAy(struct gwig *pWig, double *Xvec, double *pay, double *paypx)
+{
+  int    i;
+  double x, y, z;
+  double kx, ky, kz, tz, kw;
+  double cx, sx, chx, shx;
+  double cy, syky, chy, shy, sz;
+  double gamma0, beta0;
+  double ay, aypx;
+
+  x = Xvec[0];
+  y = Xvec[2];
+  z = pWig->Zw;
+    
+  kw   = 2e0*PI/(pWig->Lw);
+  ay   = 0e0;
+  aypx = 0e0;
+
+  gamma0  = pWig->E0/XMC2;
+  beta0   = sqrt(1e0 - 1e0/(gamma0*gamma0));
+  pWig->Aw = (q_e/m_e/clight)/(2e0*PI) * (pWig->Lw) * (pWig->PB0);
+     
+  /* Horizontal Wiggler: note that one potentially could have: kx=0 */
+  for ( i = 0; i < pWig->NHharm; i++ ){
+    pWig->HCw[i] = (pWig->HCw_raw[i])*(pWig->Aw)/(gamma0*beta0);
+    kx = pWig->Hkx[i];
+    ky = pWig->Hky[i];
+    kz = pWig->Hkz[i];
+    tz = pWig->Htz[i];
+  
+    sx = sin(kx * x);
+    shy = sinh(ky * y);
+    sz  = sin(kz * z + tz);
+    ay  = ay + (pWig->HCw[i])*(kw/kz)*(kx/ky)*sx*shy*sz;
+  
+    cx  = cos(kx * x);
+    chy = cosh(ky * y);
+    
+    aypx = aypx + (pWig->HCw[i])*(kw/kz)*pow(kx/ky,2) * cx*chy*sz;
+  }
+
+  /* Vertical Wiggler: note that one potentially could have: ky=0 */
+  for (i = 0; i < pWig->NVharm; i++) {
+    pWig->VCw[i] = (pWig->VCw_raw[i])*(pWig->Aw)/(gamma0*beta0);       
+    kx = pWig->Vkx[i];
+    ky = pWig->Vky[i];
+    kz = pWig->Vkz[i];
+    tz = pWig->Vtz[i];
+
+    chx = cosh(kx * x);
+    cy  = cos(ky * y);
+    sz  = sin(kz * z + tz);
+    ay  = ay + (pWig->VCw[i])*(kw/kz)*chx*cy*sz;
+
+    shx = sinh(kx * x);
+    if (abs(ky/kw) > GWIG_EPS) {
+      syky  = sin(ky * y)/ky;
+    } else {
+      syky = y * sinc(ky * y);
+      aypx = aypx + (pWig->VCw[i])*(kw/kz)* kx*shx*syky*sz;
+    }
+  }
+  
+  *pay = ay;
+  *paypx = aypx;
+}
+
+
+double sinc(double x)
+{
+  double x2, result;
+/* Expand sinc(x) = sin(x)/x to x^8 */
+  x2 = x*x;
+  result = 1e0 - x2/6e0*(1e0 - x2/20e0 *(1e0 - x2/42e0*(1e0-x2/72e0) ) );
+  return result;
+}
+
+
+
+void GWigB(struct gwig *pWig, double *Xvec, double *B) 
+/* Compute magnetic field at particle location.
+ * Added by M. Borland, August 2007.
+ */
+{
+  int    i;
+  double x, y, z;
+  double kx, ky, kz, tz, kw;
+  double cx, sx, chx, shx;
+  double cy, sy, chy, shy;
+  double cz;
+  double B0;

```

Comment:
Here I get a compile-time error on MacOS and python 3.6:

```
/Volumes/machfs/machine/at/pyat/integrator-src/gwigR.c:353:10: error: expected identifier or '('
  double B0;
         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/sys/termios.h:291:17: note: expanded from macro 'B0'
#define B0      0
```
Apparently the `termios.h` header is requested by python 3.6 and causes this crazy `#define B0      0` definition.
One could either put a line
`#undef B0
`
in the file, which may look strange indeed, or change the variable name (`PB0`?)